nms/app/packages/magmalte/app/components/cwf/CWFGateways.js

```diff
@@ -282,4 +255,87 @@ function CWFGateways(props: WithAlert & {}) {
   );
 }
 
+function GatewayRow(props: {
+  gateway: cwf_gateway,
+  haPairs: cwf_ha_pair[],
+}) {
+  const {gateway, haPairs} = props;
+  const [expanded, setExpanded] = useState<Set<string>>(new Set());
+  const classes = useStyles();
+  const {match, history, relativePath, relativeUrl} = useRouter();
+
+  const gatewayHaPair = haPairs.filter(haPair => {
+    return haPair.gateway_id_1 === gateway.id ||
+        haPair.gateway_id_2 === gateway.id
+  });
+
+  const isGatewayInHaPair = gatewayHaPair.length > 0;
+  const isPrimary = isGatewayInHaPair &&
+      gatewayHaPair[0]?.state?.ha_pair_status?.active_gateway === gateway.id;
+  const isGateway1 = isGatewayInHaPair &&
+      gatewayHaPair[0].gateway_id_1 === gateway.id;
+
+  const isNonHaGatewayHealthy = Math.max(0, Date.now() -
+      (gateway.status?.checkin_time || 0)) < FIVE_MINS &&
```

Comment:
Quick question here. I have similar code in FWA. I guess I had also added this check earlier. However recently when I checked, I saw that health interval is configurable. Should we use that instead of hardcoding this to FIVE_MINS.

Reply:
Are you referring to the checkin interval? We can do something like `5 * gateway?.magmad?.checkin_interval` instead of hard-coding this. In practice, I don't think we ever use anything other than 60 seconds, but this would still be better I think.

Reply:
Oh ok. Then in that case you can leave it as is. I believe I also have hardcoded 5 mins all over the place. I can cleanup everything in a single PR.